src/pages/chainlink-functions/resources/add-functions-to-projects.md

```diff
@@ -0,0 +1,317 @@
+whatsnext:
+  {
+    "Try out the Chainlink Functions Tutorials": "/chainlink-functions/tutorials/",
+    "Read the Concepts page to learn about the core concepts behind Chainlink Functions": "/chainlink-functions/resources/concepts/",
+    "Read the Architecture to understand how Chainlink Functions operates": "/chainlink-functions/resources/architecture/",
+  }
+setup: |
+  import CodeSample from "@components/CodeSample/CodeSample.astro"
+  import ChainlinkFunctions from "@features/chainlink-functions/common/ChainlinkFunctions.astro"
+---
+
+The [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit) lets you run several example requests, but extra steps are required to add Chainlink Functions to your existing projects.
+
+If you are new to Chainlink Functions, complete the steps in the [Getting Started Guide](/chainlink-functions/getting-started) to learn the basics.
+
+Using Chainlink Functions to your existing projects requires the following components:
+
+- **A consumer contract:** Use this contract to send requests to the Chainlink Functions decentralized oracle network (DON). The consumer contract imports the following dependencies:
+  - [FunctionsClient.sol](https://github.com/smartcontractkit/functions-hardhat-starter-kit/blob/main/contracts/dev/functions/FunctionsClient.sol): This includes several functions necessary to submit requests to the DON.
+  - [ConfirmedOwner.sol](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/ConfirmedOwner.sol) from the [@chainlink/contracts](https://www.npmjs.com/package/@chainlink/contracts) NPM package.
+- **A Chainlink Functions subscription:** The subscription is used to pay for Chainlink Functions requests when they are fulfilled. You can [create and fund](/chainlink-functions/resources/subscriptions) subscriptions using the starter kit tools, but this guide shows you how to manage subscriptions programmatically.
+
+## Before you begin
+
+This guide assumes you are using a [Hardhat](https://hardhat.org/) JavaScript project with Node.js, but you can modify it to work with other frameworks.
+
+You must have a wallet with testnet LINK and native tokens to run this example. This example uses Polygon Mumbai. You can get testnet LINK at [faucets.chain.link](https://faucets.chain.link/mumbai) and testnet MATIC at the [Polygon Faucet](https://faucet.polygon.technology/). To get testnet funds for other networks, see the [LINK Token Contracts](/resources/link-token-contracts) page.
+
+:::note[Request Access]
+Chainlink Functions is currently in a limited BETA.
+Apply [here](http://functions.chain.link/) to add your EVM account address to the Allow List.
+:::
+
+### Configure Hardhat
+
+If you already have a project, you can skip to the [Libraries and Dependencies](#get-the-libraries-and-contract-dependencies) section. If you don't already have a Hardhat project, start one with the following steps:
```

Comment:
can we move this disclaimer about 'if you have a project already, skip ahead' further up to be before the configure hardhat section? I feel it belongs there more?

Reply:
Moved up to the Before you begin section. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0